### PR TITLE
Add API Dummy

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Using data from #6.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+version: '3'
+
+services:
+    api-dummy-fpm:
```
Comment (giorgiosironi) [inline docker-compose.yml:4]: our current api-dummy ended up using the built-in web server because of the ease of use (only one image to specify, no configuration); an nginx and fpm setup is more realistic, but a dummy is not necessarily realistic?

Comment (thewilkybarkid) [reply docker-compose.yml:4]: The skeleton doesn't need caching etc, but multiple requests would become synchronous, making the Journal appear slow.

It's not that much extra to add here too.

Comment (giorgiosironi) [reply docker-compose.yml:4]: Skeleton-wise I think it's very good to have both nginx and fpm, since it's supposed to be touching roughly all pieces

File: api-dummy/.docker/nginx-server.conf [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        fastcgi_param DOCUMENT_ROOT $realpath_root;
+
+        internal;
+    }
+
+    location ~ \.php$ {
+        return 404;
+    }
+
+    error_log /var/log/nginx/app_error.log;
```
Comment (giorgiosironi) [inline api-dummy/.docker/nginx-server.conf]: in the default image logs probably go to stderr, which makes `docker logs` collect them; may be unnecessary to override these two

File: api-dummy/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
+FROM php:7.1-fpm
```
Comment (giorgiosironi) [inline api-dummy/Dockerfile:1]: We can't really port the elife base images out here as they contain specific software like New Relic support

File: api-dummy/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+RUN mkdir -p /var/www && \
+    chown www-data:www-data /var/www
```
Comment (giorgiosironi) [inline api-dummy/Dockerfile:4]: a user to own the files making them not writable from `www-data` would be good, but can be easily postponed

File: api-dummy/bin/console [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+set_time_limit(0);
+
+require __DIR__.'/../vendor/autoload.php';
+
+$input = new ArgvInput();
+$env = $input->getParameterOption(['--env', '-e'], $_SERVER['APP_ENV'] ?? 'dev', true);
+$debug = (bool) ($_SERVER['APP_DEBUG'] ?? ('prod' !== $env)) && !$input->hasParameterOption('--no-debug', true);
+
+if ($debug) {
+    umask(0000);
```
Comment (giorgiosironi) [inline api-dummy/bin/console]: why is this necessary? All files and directories should be created by a single user at the moment

File: api-dummy/composer.json [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        "symfony/console": "^4.0",
+        "symfony/dependency-injection": "^4.0",
+        "symfony/flex": "^1.0",
+        "symfony/framework-bundle": "^4.0",
+        "symfony/http-foundation": "^4.0",
+        "symfony/http-kernel": "^4.0",
+        "symfony/monolog-bundle": "^3.2",
+        "symfony/routing": "^4.0",
+        "symfony/yaml": "^4.0"
+    },
```
Comment (giorgiosironi) [inline api-dummy/composer.json:18]: I assume this is pure Symfony 4. Wonder about the size of `vendor/` and how it compares to the current Silex version it is substituting?

File: api-dummy/config/packages/prod/monolog.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
+monolog:
```
Comment (giorgiosironi) [inline api-dummy/config/packages/prod/monolog.yaml:1]: There is an assumption about which are the standard environments supported: `dev`, `prod`, what else?

File: api-dummy/public/index.php [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+use Symfony\Component\Debug\Debug;
+use Symfony\Component\HttpFoundation\Request;
+
+require __DIR__.'/../vendor/autoload.php';
+
+$env = $_SERVER['APP_ENV'] ?? 'dev';
+$debug = (bool) ($_SERVER['APP_DEBUG'] ?? ('prod' !== $env));
+
+if ($debug) {
+    umask(0000);
```
Comment (giorgiosironi) [inline api-dummy/public/index.php]: similar to the console's `umask`

File: api-dummy/symfony.lock [hunk 9]
```diff
@@ -0,0 +1,98 @@
+{
```
Comment (giorgiosironi) [inline api-dummy/symfony.lock:1]: Symfony Flex is a candidate for knowledge sharing

File: api-dummy/bin/console (hunk 5 above)
Comment (thewilkybarkid) [reply api-dummy/bin/console]: Just Symfony boilerplate.

File: api-dummy/config/packages/prod/monolog.yaml (hunk 7 above)
Comment (thewilkybarkid) [reply api-dummy/config/packages/prod/monolog.yaml:1]: Depends on the infrastructure. `ci`/`test`; we could end up with `end2end` putting all the distributed Libero projects together. Maybe a `demo` that we host?

Comment (giorgiosironi) [reply api-dummy/config/packages/prod/monolog.yaml:1]: The name of environments are good, we should only take care:
- not to put in assumptions about hostnames which we usually don't
- leave open the possibility to add new environments without forking the code (e.g. extending the Docker image and adding files)

File: api-dummy/.docker/nginx-server.conf (hunk 2 above)
Comment (giorgiosironi) [reply api-dummy/.docker/nginx-server.conf]: so I'd remove the two `error_log` and `access_log` directives

File: api-dummy/Dockerfile (hunk 4 above)
Comment (thewilkybarkid) [reply api-dummy/Dockerfile:4]: Sounds out of scope for this as a skeleton.

**Comment by thewilkybarkid** [comment]
Anything left to do?

**Review by giorgiosironi — APPROVED** [review]
I think this is ready to be merged, key risks are in other areas now

**Review by seanwiseman — APPROVED** [review]
